rust/otap-dataflow/crates/otap/src/query_engine_processor/config.rs

```diff
@@ -0,0 +1,11 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::Deserialize;
+
+/// Configuration for the [`QueryEngineProcessor`](super::QueryEngineProcessor)
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    /// the program that defines the transformation to be applied
+    pub program: String,
```

Comment:
perhaps another enum config to determine the query language supported? like Enum {KQL, SQL, etc...}

Reply:
Yes, I was planning on adding something like this when we add support for other languages. 

At present, I'm not sure a language enum would be all that's needed. OTTL for example isn't as expressive as KQL, and it also needs some identifier from outside the statement's syntax to identify the context. We might need some additional config around the OTTL statements for the processor to know what to do. Given this  ambiguity, I figured I'd defer adding this type of config until it's needed.